src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -536,4 +540,45 @@ private void drain() {
     public TopQueriesHealthStats getHealthStats() {
         return new TopQueriesHealthStats(this.topQueriesStore.size(), this.queryGrouper.getHealthStats());
     }
+
+    /**
+     * Validate the exporter delete after value
+     *
+     * @param deleteAfter exporter and reader settings
+     */
+    static void validateExporterDeleteAfter(final int deleteAfter) {
+        if (deleteAfter < MIN_DELETE_AFTER_VALUE || deleteAfter > MAX_DELETE_AFTER_VALUE) {
+            OperationalMetricsCounter.getInstance().incrementCounter(OperationalMetric.INVALID_EXPORTER_TYPE_FAILURES);
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "Invalid exporter delete_after_days setting [%d], value should be an integer between %d and %d.",
+                    deleteAfter,
+                    MIN_DELETE_AFTER_VALUE,
+                    MAX_DELETE_AFTER_VALUE
+                )
+            );
+        }
+    }
+
+    /**
+     * Set exporter delete after if exporter is a {@link LocalIndexExporter}
+     *
+     * @param deleteAfter the number of days after which Top N local indices should be deleted
+     */
+    void setExporterDeleteAfter(final int deleteAfter) {
+        if (exporter != null && exporter.getClass() == LocalIndexExporter.class) {
+            ((LocalIndexExporter) exporter).setDeleteAfter(deleteAfter);
+        }
+    }
+
+    /**
+     * Delete Top N local indices older than the configured data retention period
+     */
+    void deleteExpiredIndices(final Map<String, IndexMetadata> indexMetadataMap) {
+        if (exporter != null && exporter.getClass() == LocalIndexExporter.class) {
```

Comment:
As I think about this more, should the delete function rely on the index exporter? Cause in this case, if we switch the exporter type, there will be always be lingering indices left since we do the deletion only if `exporter.getClass() == LocalIndexExporter.class`

Reply:
That is a good point. Is it okay to perennially check for expired indices when sink type/exporter type is not local index? This involves a for loop over all index names. 

If so, delete logic should move from `LocalIndexExporter` to `TopQueriesService`